includes/class-wc-payments-checkout.php

```diff
@@ -161,6 +161,9 @@ public function get_payment_fields_js_config() {
 		// Needed to init the hooks.
 		WC_Checkout::instance();
 
+		// The registered card gateway is more reliable than $this->gateway, but if it isn't available for any reason, fall back to the gateway provided to this checkout class.
+		$gateway = WC_Payments::get_registered_card_gateway() ? WC_Payments::get_registered_card_gateway() : $this->gateway;
```

Comment:
```suggestion
		$gateway = WC_Payments::get_registered_card_gateway() ?? $this->gateway;
```

Would the null coalescing operator work directly?

Reply:
Great tip, thanks! 32c607e1b86d481546de3aadd99994476d7a7307